category_tool.py

```diff
@@ -0,0 +1,329 @@
+import os
+import random
+import re
+import string
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+
+import yaml
+
+
+class ModAction(Enum):
+    ADD = 1
+    UPDATE = 2
+    MOVE = 3
+    INACTIVATE = 4
+
+
+class CategoryStatus(Enum):
+    ACTIVE = 1
+    INACTIVE = 2
+    REPLACED = 3
+
+
+@dataclass
+class Category:
+    code: str
+    level: int
+    name: str
+    status: CategoryStatus = CategoryStatus.ACTIVE
+    description: str = ''
+    replaces: str = None
+    replacedBy: str = None
+
+
+@dataclass
+class ModOperation:
+    action: ModAction
+    code: str = None
+    description: str = None
+    name: str = None
+    newName: str = None
+    newParentCode: str = None
+    newParentName: str = None
+    parentName: str = None
+    parentCode: str = None
+
+
+def generate_new_code(parent_code, existing_codes):
+    while True:
+        sub_code = ""
+
+        for _ in range(3):
+            sub_code += random.choice(string.ascii_uppercase)
+
+        new_code = parent_code + '-' + sub_code
+        if new_code not in existing_codes:
+            break
+
+    return new_code
+
+
+version = None
+
+applied_mods = []
+
+mod_file_path = 'src/mods'
+
+
+def calculate_level(category_code):
+    return len(category_code.split('-')) - 1
+
+
+def load_categories():
+    global version
+    with open('src/categories-coded.yml', 'r', encoding='utf-8') as categories_file:
+        categories_dict = yaml.load(categories_file)
+
+    categories = {}
+
+    for category in categories_dict['categories']:
+        cat = Category(code=category['code'], level=category['level'], name=category['name'])
+
+        if 'description' in category:
+            cat.description = category['description']
+        if 'replaces' in category:
+            cat.replaces = category['replaces']
+        if 'replacedBy' in category:
+            cat.replacedBy = category['replacedBy']
+        if 'status' in category:
+            cat.status = CategoryStatus[category['status']]
+
+        categories[(cat.code, cat.name)] = cat
+
+    version = categories_dict['version']
+
+    return categories
+
+
+def get_unprocessed_modification_files():
+    global mod_file_path
+    global applied_mods
```

Comment:
I really dislike this. I'd prefer to pass the already applied mods (and mod file path) into this function, and then add the new files to it afterwards.